cli.go

```diff
@@ -101,39 +121,43 @@ func runSqs(ctx *cli.Context) error {
 		return errors.New("missing SQS queue URL")
 	}
 
-	err := setupSharedResources(ctx.Bool("debug"), ctx.String("redis-url"))
+	dbPool, err := buildRedisPool(ctx.String("redis-url"))
 	if err != nil {
 		return err
 	}
 
+	db := &redisRepo{cg: dbPool}
+	log := buildLog(ctx.Bool("debug"))
+	sqsSvc := sqs.New(session.New())
+	snsSvc := sns.New(session.New(), &aws.Config{
+		Region: aws.String(ctx.String("aws-region")),
+	})
+	asSvc := autoscaling.New(session.New(), &aws.Config{
+		Region: aws.String(ctx.String("aws-region")),
+	})
+
 	cntx, cancel := context.WithCancel(context.Background())
 	go runSignalHandler(cancel)
-	return newSqsHandler(sqsQueueURL, ctx.Int("concurrency")).Run(cntx)
-}
 
-func setupSharedResources(debug bool, redisURL string) error {
-	var err error
+	return (&sqsHandler{
+		queueURL:    sqsQueueURL,
+		concurrency: ctx.Int("concurrency"),
 
-	log = logrus.New()
+		db: db,
+		l:  log,
+		a:  asSvc,
+		n:  snsSvc,
+		s:  sqsSvc,
```

Comment:
If it were Ruby or Python or even Bash, I would agree with you.  I would argue that single-letter vars in Go are idiomatic, and don't cause the harm that they do in dynamic languages because yay types.